numba/core/types/functions.py

```diff
@@ -145,6 +145,8 @@ def format(self):
             elif (is_external_fn_ptr and
                   isinstance(source_fn, numba.core.typing.templates.Signature)):
                 source_fn = template.__class__
+            elif hasattr(template, '_overload_func'):
+                source_fn = template._overload_func
```

Comment:
perhaps we should add a method to `Template` to get the source function

Reply:
I think that would be the ideal thing to do, but am reluctant to patch that in here with view of minimising changes between 0.50.0rc1 and 0.50.0.